### PR TITLE
NetworkPkg: Additional CodeQL Fixes

### PR DESCRIPTION
**Pull request by makubacki**
## Description

Additional CodeQL fixes in NetworkPkg.

- [ ] Impacts functionality?
  - **Functionality** - Does the change ultimately impact how firmware functions?
  - Examples: Add a new library, publish a new PPI, update an algorithm, ...
- [x] Impacts security?
  - **Security** - Does the change have a direct security impact on an application,
    flow, or firmware?
  - Examples: Crypto algorithm change, buffer overflow fix, parameter
    validation improvement, ...
- [ ] Breaking change?
  - **Breaking change** - Will anyone consuming this change experience a break
    in build or boot behavior?
  - Examples: Add a new library class, move a module to a different repo, call
    a function in a new library class in a pre-existing module, ...
- [ ] Includes tests?
  - **Tests** - Does the change include any explicit test code?
  - Examples: Unit tests, integration tests, robot tests, ...
- [ ] Includes documentation?
  - **Documentation** - Does the change contain explicit documentation additions
    outside direct code modifications (and comments)?
  - Examples: Update readme file, add feature readme file, link to documentation
    on an a separate Web page, ...

## How This Was Tested

CodeQL CLI plugin

## Integration Instructions

N/A

Co-authored-by: Taylor Beebe <tabeebe@microsoft.com>
Signed-off-by: Michael Kubacki <michael.kubacki@microsoft.com>

### CODE REVIEW

**Comment by makubacki** [comment]
Note: Add `MU_CHANGE` tags is a todo

File: NetworkPkg/Application/VConfig/VConfig.c [hunk 1]
```diff
@@ -654,18 +654,40 @@ VlanConfigMain (
...
     Str = ShellCommandLineGetValue (List, L"-a");
-    AddVlan ((CHAR16 *)Str);
+    // MU_CHANGE [BEGIN] - CodeQL change
+    if (Str == NULL) {
+      ShellPrintHiiEx (-1, -1, NULL, STRING_TOKEN (STR_VCONFIG_NO_IF), mHiiHandle);
+      goto Exit;
+    }
+
+    // MU_CHANGE [END] - CodeQL change
+    DisplayVlan (NULL);
```
Comment (os-d) [inline NetworkPkg/Application/VConfig/VConfig.c]: Should this be AddVlan (deleted line 663 above) not DisplayVlan?

File: NetworkPkg/HttpDxe/HttpsSupport.c [hunk 2]
```diff
@@ -1468,9 +1485,15 @@ TlsCloseSession (
...
-  DataOut   = NetbufAllocSpace (PacketOut, (UINT32)BufferOutSize, NET_BUF_TAIL);
+  if (PacketOut == NULL) {
+    FreePool (BufferOut);
+    return EFI_OUT_OF_RESOURCES;
+  }
+
+  DataOut = NetbufAllocSpace (PacketOut, (UINT32)BufferOutSize, NET_BUF_TAIL);
   if (DataOut == NULL) {
     FreePool (BufferOut);
+    NetbufFree (PacketOut);
```
Comment (os-d) [inline NetworkPkg/HttpDxe/HttpsSupport.c:1500]: nit: MU_CHANGE tags around this one?

File: NetworkPkg/TlsAuthConfigDxe/TlsAuthConfigImpl.c [hunk 3]
```diff
@@ -1225,8 +1231,14 @@ TlsAuthConfigAccessExtractConfig (
...
-    ConfigRequest    = AllocateZeroPool (Size);
+    // MU_CHANGE [BEGIN] - CodeQL change
+    if (ConfigRequestHdr == NULL) {
+      return EFI_OUT_OF_RESOURCES;
+    }
+
+    // MU_CHANGE [END] - CodeQL change
+    Size          = (StrLen (ConfigRequestHdr) + 32 + 1) * sizeof (CHAR16);
+    ConfigRequest = AllocateZeroPool (Size);
     ASSERT (ConfigRequest != NULL);
```
Comment (os-d) [inline NetworkPkg/TlsAuthConfigDxe/TlsAuthConfigImpl.c]: Looks like CodeQL missed this one? We don't check ConfigRequest before using it.

File: NetworkPkg/VlanConfigDxe/VlanConfigImpl.c [hunk 4]
```diff
@@ -113,8 +113,14 @@ VlanExtractConfig (
...
-    Size             = (StrLen (ConfigRequestHdr) + 32 + 1) * sizeof (CHAR16);
-    ConfigRequest    = AllocateZeroPool (Size);
+    // MU_CHANGE [BEGIN] - CodeQL change
+    if (ConfigRequestHdr == NULL) {
+      return EFI_OUT_OF_RESOURCES;
+    }
+
+    // MU_CHANGE [END] - CodeQL change
+    Size          = (StrLen (ConfigRequestHdr) + 32 + 1) * sizeof (CHAR16);
+    ConfigRequest = AllocateZeroPool (Size);
```
Comment (os-d) [inline NetworkPkg/VlanConfigDxe/VlanConfigImpl.c:123]: Same here, not sure if we want to fix or just do the ones CodeQL catches.

**Review by os-d — APPROVED** [review]

**Review by kenlautner — APPROVED** [review]

File: NetworkPkg/Ip6Dxe/Ip6Mld.c [hunk 5]
```diff
@@ -297,7 +304,14 @@ Ip6SendMldDone (
   // Fill in MLD message - Done
   //
   MldHead = (IP6_MLD_HEAD *)NetbufAllocSpace (Packet, sizeof (IP6_MLD_HEAD), FALSE);
-  ASSERT (MldHead != NULL);
+  // MU_CHANGE [BEGIN] - CodeQL change
+  if (MldHead == NULL) {
```
Comment (makubacki) [inline NetworkPkg/Ip6Dxe/Ip6Mld.c:308]: Looks like `NetbufFree (Options);` should be in this block.

File: NetworkPkg/Ip6Dxe/Ip6Mld.c [hunk 6]
```diff
@@ -193,7 +193,14 @@ Ip6SendMldReport (
   // Fill in MLD message - Report
   //
   MldHead = (IP6_MLD_HEAD *)NetbufAllocSpace (Packet, sizeof (IP6_MLD_HEAD), FALSE);
-  ASSERT (MldHead != NULL);
+  // MU_CHANGE [BEGIN] - CodeQL change
+  if (MldHead == NULL) {
```
Comment (makubacki) [inline NetworkPkg/Ip6Dxe/Ip6Mld.c:197]: Looks like `NetbufFree (Options);` should be in this block.

File: NetworkPkg/Ip6Dxe/Ip6Nd.c [hunk 7]
```diff
@@ -1169,7 +1176,14 @@ Ip6SendRouterSolicit (
                                                  sizeof (IP6_ETHER_ADDR_OPTION),
                                                  FALSE
                                                  );
-    ASSERT (LinkLayerOption != NULL);
+    // MU_CHANGE [BEGIN] - CodeQL change
+    if (LinkLayerOption == NULL) {
```
Comment (makubacki) [inline NetworkPkg/Ip6Dxe/Ip6Nd.c:1180]: Looks like `NetBufFree (IcmpHead);` should be in this block.

File: NetworkPkg/Ip6Dxe/Ip6Nd.c [hunk 8]
```diff
@@ -1281,15 +1302,29 @@ Ip6SendNeighborAdvertise (
   }
 
   Target = (EFI_IPv6_ADDRESS *)NetbufAllocSpace (Packet, sizeof (EFI_IPv6_ADDRESS), FALSE);
-  ASSERT (Target != NULL);
+  // MU_CHANGE [BEGIN] - CodeQL change
+  if (Target == NULL) {
```
Comment (makubacki) [inline NetworkPkg/Ip6Dxe/Ip6Nd.c:1306]: Looks like `NetBufFree (IcmpHead);` should be in this block.

Comment (makubacki) [reply NetworkPkg/Ip6Dxe/Ip6Nd.c:1306]: Same in block below in same function.

File: NetworkPkg/Ip6Dxe/Ip6Nd.c [hunk 9]
```diff
@@ -1281,15 +1302,29 @@ Ip6SendNeighborAdvertise (
...
   IP6_COPY_ADDRESS (Target, TargetIp6Address);
 
   LinkLayerOption = (IP6_ETHER_ADDR_OPTION *)NetbufAllocSpace (
                                                Packet,
                                                sizeof (IP6_ETHER_ADDR_OPTION),
                                                FALSE
                                                );
-  ASSERT (LinkLayerOption != NULL);
+  // MU_CHANGE [BEGIN] - CodeQL change
+  if (LinkLayerOption == NULL) {
```
Comment (makubacki) [inline NetworkPkg/Ip6Dxe/Ip6Nd.c:1321]: Also add `NetBufFree (Target);` here.

File: NetworkPkg/Ip6Dxe/Ip6Nd.c [hunk 10]
```diff
@@ -1416,7 +1458,14 @@ Ip6SendNeighborSolicit (
                                                  sizeof (IP6_ETHER_ADDR_OPTION),
                                                  FALSE
                                                  );
-    ASSERT (LinkLayerOption != NULL);
+    // MU_CHANGE [BEGIN] - CodeQL change
+    if (LinkLayerOption == NULL) {
```
Comment (makubacki) [inline NetworkPkg/Ip6Dxe/Ip6Nd.c:1462]: `NetBufFree (IcmpHead);` in this block.

File: NetworkPkg/TlsAuthConfigDxe/TlsAuthConfigImpl.c (hunk 3 above)
Comment (makubacki) [reply NetworkPkg/TlsAuthConfigDxe/TlsAuthConfigImpl.c]: Thanks. I'll update it as well.

File: NetworkPkg/VlanConfigDxe/VlanConfigImpl.c (hunk 4 above)
Comment (makubacki) [reply NetworkPkg/VlanConfigDxe/VlanConfigImpl.c:123]: I'll update it as well.

File: NetworkPkg/HttpDxe/HttpsSupport.c (hunk 2 above)
Comment (makubacki) [reply NetworkPkg/HttpDxe/HttpsSupport.c:1500]: Update in PR commit 93db58d439.

File: NetworkPkg/Application/VConfig/VConfig.c (hunk 1 above)
Comment (makubacki) [reply NetworkPkg/Application/VConfig/VConfig.c]: Updated in PR commit 9108f8b6b0.

File: NetworkPkg/Ip6Dxe/Ip6Mld.c (hunk 5 above)
Comment (makubacki) [reply NetworkPkg/Ip6Dxe/Ip6Mld.c:308]: These `NetbufAllocSpace()` calls appear to be carving out space from an already allocated buffer so I'll drop my comment on them.

File: NetworkPkg/Ip6Dxe/Ip6Nd.c (hunk 7 above)
Comment (makubacki) [reply NetworkPkg/Ip6Dxe/Ip6Nd.c:1180]: These `NetbufAllocSpace()` calls appear to be carving out space from an already allocated buffer so I'll drop my comment on them.

File: NetworkPkg/Ip6Dxe/Ip6Nd.c (hunk 8 above)
Comment (makubacki) [reply NetworkPkg/Ip6Dxe/Ip6Nd.c:1306]: These `NetbufAllocSpace()` calls appear to be carving out space from an already allocated buffer so I'll drop my comment on them.

File: NetworkPkg/Ip6Dxe/Ip6Nd.c (hunk 9 above)
Comment (makubacki) [reply NetworkPkg/Ip6Dxe/Ip6Nd.c:1321]: These `NetbufAllocSpace()` calls appear to be carving out space from an already allocated buffer so I'll drop my comment on them.

File: NetworkPkg/TlsAuthConfigDxe/TlsAuthConfigImpl.c (hunk 3 above)
Comment (makubacki) [reply NetworkPkg/TlsAuthConfigDxe/TlsAuthConfigImpl.c]: Done in PR commit 4130250690.

File: NetworkPkg/VlanConfigDxe/VlanConfigImpl.c (hunk 4 above)
Comment (makubacki) [reply NetworkPkg/VlanConfigDxe/VlanConfigImpl.c:123]: Done in PR commit e62511c933.

File: NetworkPkg/Ip6Dxe/Ip6Mld.c (hunk 6 above)
Comment (makubacki) [reply NetworkPkg/Ip6Dxe/Ip6Mld.c:197]: These `NetbufAllocSpace()` calls appear to be carving out space from an already allocated buffer so I'll drop my comment on them.

File: NetworkPkg/Ip6Dxe/Ip6Nd.c (hunk 10 above)
Comment (makubacki) [reply NetworkPkg/Ip6Dxe/Ip6Nd.c:1462]: These `NetbufAllocSpace()` calls appear to be carving out space from an already allocated buffer so I'll drop my comment on them.

**Pull request merged**